include/common/thread.h

```diff
@@ -5,6 +5,40 @@
 #include <thread>
 
 namespace BoBRobotics {
+//! A simple RAII wrapper for std::thread
```

Comment:
Presumably this requires the thread to implement some sort of signalling to kill it before you join? I kinda feel the RAAIness just makes it easier to create a deadlock on exit

Reply:
You could end up with a deadlock, but the problem with C++ threads is if you just let them be destroyed without ``join()``ing or ``deatch()``ing them then the behaviour is undefined. I took this idea from Scott Meyers' book.

Reply:
I dunno - I think on any platform with an OS (rather than bare metal), the OS's going to kill any lingering threads - in the same way it frees any outstanding heap allocations - bad practise but imo maybe a better default than deadlocking